titus-server-runtime/src/main/java/com/netflix/titus/runtime/connector/registry/DefaultDockerRegistryClient.java

```diff
@@ -0,0 +1,129 @@
+public class DefaultDockerRegistryClient implements RegistryClient {
+
+    private static final Logger logger = LoggerFactory.getLogger(DefaultDockerRegistryClient.class);
+
+    private static final String acceptHeader = "Accept";
+    private static final String dockerManifestType = "application/vnd.docker.distribution.manifest.v2+json";
+    private static final String dockerDigestHeaderKey = "Docker-Content-Digest";
+
+    private static final Map<String, String> headers = Collections.unmodifiableMap(
+            Stream.of(new AbstractMap.SimpleEntry<>(acceptHeader, dockerManifestType))
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue))
+    );
+
+    private final TitusRegistryClientConfiguration titusRegistryClientConfiguration;
+    private final RxRestClient restClient;
+
+    @Inject
+    DefaultDockerRegistryClient(RegistryEndpointResolver endpointResolver, TitusRegistryClientConfiguration configuration, Registry spectatorRegistry) {
+        this.titusRegistryClientConfiguration = configuration;
+
+        RxRestClients.Builder builder = RxRestClients.newBuilder("dockerRegistryClient", spectatorRegistry)
+                .endpointResolver(endpointResolver)
+                // We rely on the RxRestClient for timeout handling
+                .timeUnit(TimeUnit.MILLISECONDS)
+                .requestTimeout(titusRegistryClientConfiguration.getRegistryTimeoutMs())
+                .retryCount(titusRegistryClientConfiguration.getRegistryRetryCount())
+                .retryDelay(titusRegistryClientConfiguration.getRegistryRetryDelayMs())
+                .noRetryStatuses(new HashSet<>(Collections.singletonList(HttpResponseStatus.NOT_FOUND)));
+        if (titusRegistryClientConfiguration.isSecure()) {
+            try {
+                builder.sslEngineFactory(new NettySslContextEngineFactory(SslContextBuilder.forClient().build()));
+            } catch (SSLException e) {
+                logger.error("Unable configure Docker registry client SSL context: {}", e.getMessage());
+                throw new RuntimeException("Error configuring SSL context", e);
+            }
+        }
+        this.restClient = builder.build();
+    }
+
+    /**
+     * Gets the Docker Version 2 Schema 2 Content Digest for the provide repository and reference. The
+     * reference may be an image tag or digest value. If the image does not exist or another error is
+     * encountered, an onError value is emitted.
+     */
+    public Single<String> getImageDigest(String repository, String reference) {
+        return registryRequestWithErrorHandling(restClient.doGET(buildRegistryUri(repository, reference), headers, TypeProviders.ofEmptyResponse()))
+                .map(RxHttpResponse::getHeaders)
+                .flatMap(stringListMap -> {
+                    if (stringListMap.containsKey(dockerDigestHeaderKey)) {
+                        return Observable.from(stringListMap.get("Docker-Content-Digest"));
```

Comment:
use constant `dockerDigestHeaderKey` defined above

Reply:
Yeah. No idea why I retyped the whole thing...Will fix.